pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -143,38 +143,35 @@ type reconciler struct {
 }
 
 func (rc *reconciler) Run(stopCh <-chan struct{}) {
-	wait.Until(rc.reconciliationLoopFunc(), rc.loopSleepDuration, stopCh)
-}
-
-func (rc *reconciler) reconciliationLoopFunc() func() {
-	return func() {
-		rc.reconcile()
-
-		// Sync the state with the reality once after all existing pods are added to the desired state from all sources.
-		// Otherwise, the reconstruct process may clean up pods' volumes that are still in use because
-		// desired state of world does not contain a complete list of pods.
-		if rc.populatorHasAddedPods() && !rc.StatesHasBeenSynced() {
-			klog.InfoS("Reconciler: start to sync state")
-			rc.sync()
-		}
-	}
+	go rc.sync()
```

Comment:
So basically reconcile can run parallel to reconstruction and we do not wait for DSOW to be populate before doing reconstruction? This means for any volume type, where reconstruction fails, the volume/mount point may leak.

Reply:
I see, reconstruction needs DSW populated here to force unmount volumes where the reconstruction itself failed: https://github.com/kubernetes/kubernetes/blob/f0d5ea1e1d4d43cd5caac92800207fcd2a62538c/pkg/kubelet/volumemanager/reconciler/reconciler.go#L377-L390

So, back to the drawing board. I was thinking about adding just an "uncertain tombstone" to ASW, where the reconstruction sync would just mark that there is a dir for the volume in /var/lib/kubelet/pods and the actual reconstruction would happen in UnmountVolume / MountVolume operations. We could try reconstruction couple of times and if it fails for too long, then force unmount. IMO it would be cleaner that today's approach (try once), but it's also more complicated.

Reply:
>  I was thinking about adding just an "uncertain tombstone" to ASW, where the reconstruction sync would just mark that there is a dir for the volume in /var/lib/kubelet/pods and the actual reconstruction would happen in UnmountVolume / MountVolume operations

Tried that, failed fast. To mark anything as uncertain, kubelet needs UniqueVolumeID and that is not available before reconstruction completes.